flashinfer/quantization/kernels/cutile/per_token_group_quant_8bit_cutile.py

```
@@ -0,0 +1,298 @@
+        num_tokens = x.shape[-2] if x.dim() >= 2 else x.shape[0]
+        if scale_tma_aligned:
+            # TMA-friendly layout: (num_groups, aligned_num_tokens), align to 4 floats (16B)
+            aligned_size = _ceil_align(num_tokens, 4)
+            x_s_raw = torch.zeros(
+                (num_groups, aligned_size),
+                device=x.device,
+                dtype=torch.float32,
+            )
+            x_s_col_stride = aligned_size
+        else:
+            shape = (num_groups,) + x.shape[:-1]
```

Ruff (0.15.21)

[warning] Consider `(num_groups, *x.shape[:-1])` instead of concatenation

Replace with `(num_groups, *x.shape[:-1])`

(RUF005)